src/content/app/genome-browser/state/browser-location/browserLocationSlice.ts

```diff
@@ -0,0 +1,173 @@
+  regionEditorActive: false,
+  regionFieldActive: false,
+  isObjectInDefaultPosition: false
+};
+
+export const setActualChrLocation: ActionCreator<
+  ThunkAction<any, any, null, Action<string>>
+> = (chrLocation: ChrLocation) => {
+  return (dispatch: Dispatch, getState: () => RootState) => {
+    const state = getState();
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+    if (!activeGenomeId) {
+      return;
+    }
+    const payload = {
+      [activeGenomeId]: chrLocation
+    };
+
+    dispatch(updateActualChrLocation(payload));
+  };
+};
+
+export const setChrLocation: ActionCreator<
+  ThunkAction<any, any, null, Action<string>>
+> = (chrLocation: ChrLocation) => {
+  return (dispatch: Dispatch, getState: () => RootState) => {
+    const state = getState();
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+    const activeEnsObjectId = getBrowserActiveEnsObjectId(state);
+    const savedChrLocation = getChrLocation(state);
+    if (!activeGenomeId || !activeEnsObjectId) {
+      return;
+    }
+    const payload = {
+      [activeGenomeId]: chrLocation
+    };
+
+    dispatch(updateChrLocation(payload));
+    browserStorageService.updateChrLocation(payload);
+
+    if (!isEqual(chrLocation, savedChrLocation)) {
+      const newUrl = urlFor.browser({
+        genomeId: activeGenomeId,
+        focus: buildFocusIdForUrl(activeEnsObjectId),
+        location: getChrLocationStr(chrLocation)
+      });
+      dispatch(replace(newUrl));
+    }
+  };
+};
+
+const browserLocationSlice = createSlice({
+  name: 'genome-browser-location',
+  initialState: defaultBrowserLocationState as BrowserLocationState,
+  reducers: {
+    setLocationDataFromUrl(state, action: PayloadAction<ParsedUrlPayload>) {
+      const { activeGenomeId, chrLocation } = action.payload;
+      if (chrLocation) {
+        state.chrLocations[activeGenomeId] = chrLocation;
+      } else {
+        state.chrLocations = pickBy(
+          state.chrLocations,
+          (value, key) => key !== activeGenomeId
+        );
+      }
+    },
+    updateChrLocation(state, action: PayloadAction<ChrLocations>) {
+      state.chrLocations = {
+        ...state.chrLocations,
+        ...action.payload
+      };
+    },
+    updateActualChrLocation(state, action: PayloadAction<ChrLocations>) {
+      state.actualChrLocations = {
+        ...state.actualChrLocations,
+        ...action.payload
+      };
+    },
+    toggleRegionEditorActive(state, action: PayloadAction<boolean>) {
+      state.regionEditorActive = action.payload;
+    },
+    toggleRegionFieldActive(state, action: PayloadAction<boolean>) {
+      state.regionFieldActive = action.payload;
+    },
+    updateDefaultPositionFlag(state, action: PayloadAction<boolean>) {
+      state.isObjectInDefaultPosition = action.payload;
+    },
+    deleteGenomeLocationData(state, action: PayloadAction<string>) {
+      const genomeId = action.payload;
+      state.chrLocations = pickBy(
```

Comment:
`delete state.chrLocations[genomeId]`.

same below

Reply:
That's done